### PR TITLE
Tracing: add support for file permissions

### PR DESCRIPTION
**Pull request by Jianlin-lv**
For the observed file/directory, add the associated permission and output the complete file/directory permission information, consistent with the format of 'ls -l'. Can be used in user space to further filter/identify events based on specific file attributes.

### CODE REVIEW

**Review by tpapagian — COMMENTED** [review]
Thanks for contributing to Tetragon! I have tried to run that and I got an event like the following:
```
{
  "process_kprobe": {
    "process": {
      "exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjYwMzI3ODcyMDgwNzU0OjMzMDk3MA==",
      "pid": 330970,
      "uid": 1010,
      "cwd": "/home/test",
      "binary": "/usr/bin/chmod",
      "arguments": "0777 a.txt"",
      "flags": "execve clone",
      "start_time": "2024-03-15T08:35:25.447770357Z",
      "auid": 1010,
      "parent_exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjU4NTYzOTcwMDAwMDAwOjI4Njc0MQ==",
      "refcnt": 1,
      "tid": 330970
    },
    "parent": {
      "exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjU4NTYzOTcwMDAwMDAwOjI4Njc0MQ==",
      "pid": 286741,
      "uid": 1010,
      "cwd": "/home/test/tetragon",
      "binary": "/usr/bin/bash",
      "flags": "procFS auid",
      "start_time": "2024-03-15T08:06:01.545689544Z",
      "auid": 1010,
      "parent_exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjU4NTYzODEwMDAwMDAwOjI4Njc0MA==",
      "tid": 286741
    },
    "function_name": "security_path_chmod",
    "args": [
      {
        "path_arg": {
          "path": "/home/test/a.txt",
          "permission": "-rw-rw-r--"
        }
      }
    ],
    "action": "KPROBE_ACTION_POST",
    "policy_name": "permission-test",
    "return_action": "KPROBE_ACTION_POST"
  },
  "node_name": "ubuntu-dev",
  "time": "2024-03-15T08:35:25.448337205Z"
}
```
and this is great!

On the other hand, it seems that some tests are failing. Maybe you can try to run `make tests` locally to check what exactly fails and try to fix that. Thanks!

**Comment by Jianlin-lv** [comment]
> Thanks for contributing to Tetragon! I have tried to run that and I got an event like the following:
> 
> ```
> {
>   "process_kprobe": {
>     "process": {
>       "exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjYwMzI3ODcyMDgwNzU0OjMzMDk3MA==",
>       "pid": 330970,
>       "uid": 1010,
>       "cwd": "/home/test",
>       "binary": "/usr/bin/chmod",
>       "arguments": "0777 a.txt"",
>       "flags": "execve clone",
>       "start_time": "2024-03-15T08:35:25.447770357Z",
>       "auid": 1010,
>       "parent_exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjU4NTYzOTcwMDAwMDAwOjI4Njc0MQ==",
>       "refcnt": 1,
>       "tid": 330970
>     },
>     "parent": {
>       "exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjU4NTYzOTcwMDAwMDAwOjI4Njc0MQ==",
>       "pid": 286741,
>       "uid": 1010,
>       "cwd": "/home/test/tetragon",
>       "binary": "/usr/bin/bash",
>       "flags": "procFS auid",
>       "start_time": "2024-03-15T08:06:01.545689544Z",
>       "auid": 1010,
>       "parent_exec_id": "YXBhcGFnLXVidW50dS1kZXY6MjU4NTYzODEwMDAwMDAwOjI4Njc0MA==",
>       "tid": 286741
>     },
>     "function_name": "security_path_chmod",
>     "args": [
>       {
>         "path_arg": {
>           "path": "/home/test/a.txt",
>           "permission": "-rw-rw-r--"
>         }
>       }
>     ],
>     "action": "KPROBE_ACTION_POST",
>     "policy_name": "permission-test",
>     "return_action": "KPROBE_ACTION_POST"
>   },
>   "node_name": "ubuntu-dev",
>   "time": "2024-03-15T08:35:25.448337205Z"
> }
> ```
> 
> and this is great!
> 
> On the other hand, it seems that some tests are failing. Maybe you can try to run `make tests` locally to check what exactly fails and try to fix that. Thanks!

hi ,  fixed  test failed issue locally.    How can I trigger checks  on this PR?

File: pkg/k8s/apis/cilium.io/v1alpha1/types.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ type KProbeArg struct {
 	// +kubebuilder:validation:Minimum=0
 	// Position of the argument.
 	Index uint32 `json:"index"`
-	// +kubebuilder:validation:Enum=auto;int;int8;uint8;int16;uint16;uint32;int32;uint64;int64;char_buf;char_iovec;size_t;skb;sock;string;fd;file;filename;path;nop;bpf_attr;perf_event;bpf_map;user_namespace;capability;kiocb;iov_iter;cred;load_info;module;syscall64;kernel_cap_t;cap_inheritable;cap_permitted;cap_effective;linux_binprm;data_loc;net_device
+	// +kubebuilder:validation:Enum=auto;int;int8;uint8;int16;uint16;uint32;int32;uint64;int64;char_buf;char_iovec;size_t;skb;sock;string;fd;file;filename;path;permission;nop;bpf_attr;perf_event;bpf_map;user_namespace;capability;kiocb;iov_iter;cred;load_info;module;syscall64;kernel_cap_t;cap_inheritable;cap_permitted;cap_effective;linux_binprm;data_loc;net_device
```
Comment (tpapagian) [inline pkg/k8s/apis/cilium.io/v1alpha1/types.go:58]: I believe that we don't need that change. Let me provide my understanding and correct me if I am wrong. An example tracing policy for your PR is:
```
apiVersion: cilium.io/v1alpha1
kind: TracingPolicy
metadata:
  name: "path-chmod"
spec:
  kprobes:
  - call: "security_path_chmod"
    syscall: false
    args:
    - index: 0
      type: "path"
    - index: 1
      type: "uint16"
```
So what you implemented here adds a new permission field if the user defines a type which is "path", "file", or some other values. But the user cannot specify `type: "permission"`. This is the list of possible types, so you can remove `permission` from that list (and all autogenerated code).

File: pkg/k8s/apis/cilium.io/v1alpha1/version.go [hunk 2]
```diff
@@ -7,4 +7,4 @@ package v1alpha1
 // Used to determine if CRD needs to be updated in cluster
 //
 // Developers: Bump patch for each change in the CRD schema.
-const CustomResourceDefinitionSchemaVersion = "1.1.8"
+const CustomResourceDefinitionSchemaVersion = "1.1.9"
```
Comment (tpapagian) [inline pkg/k8s/apis/cilium.io/v1alpha1/version.go:10]: Please also remove that change in the case where you remove any changes from `types.go`.

**Review by tpapagian — COMMENTED** [review]
Thanks for taking the time to fix the tests. All seems fine now. 

So at a high level you modify `copy_path` kernel function which is called when we have the following types:
* [kiocb_type](https://github.com/cilium/tetragon/blob/297845632936cf2116021a33e34b18f352aae9cf/bpf/process/types/basic.h#L2527)
* [file_ty](https://github.com/cilium/tetragon/blob/297845632936cf2116021a33e34b18f352aae9cf/bpf/process/types/basic.h#L2536)
* [path_ty](https://github.com/cilium/tetragon/blob/297845632936cf2116021a33e34b18f352aae9cf/bpf/process/types/basic.h#L2542)
* [linux_binprm_type](https://github.com/cilium/tetragon/blob/297845632936cf2116021a33e34b18f352aae9cf/bpf/process/types/basic.h#L2579C7-L2579C24)

In the users-space (in `args.go`) you handle the first three types but you miss the last one which is handled [here](https://github.com/cilium/tetragon/blob/297845632936cf2116021a33e34b18f352aae9cf/pkg/sensors/tracing/args.go#L499). I believe that you have also to read `mode` there as well.

Furthermore, I have one more comment regarding the tracing policy types.

Let me know if any of these does not make sense. Thanks!

File: pkg/k8s/apis/cilium.io/v1alpha1/version.go (hunk 2 above)
Comment (Jianlin-lv) [reply pkg/k8s/apis/cilium.io/v1alpha1/version.go]: Done

File: pkg/k8s/apis/cilium.io/v1alpha1/types.go (hunk 1 above)
Comment (Jianlin-lv) [reply pkg/k8s/apis/cilium.io/v1alpha1/types.go]: Thanks for your comments,  I have update  this PR. Pls take a look.

**Review by tpapagian — APPROVED** [review]
Thanks for taking the time to make the requested changes. This LGTM now!

**Pull request merged**